### PR TITLE
UNR-2702 Added a readiness check for EntityPool

### PR DESCRIPTION
**Pull request by aleximprobable**
**Contributions**: We are not currently taking public contributions - see our [contributions](CONTRIBUTING.md) policy. However, we are accepting issues and we do want your [feedback](../README.md#give-us-feedback).

-------

#### Description
Added a readiness check for EntityPool

#### Primary reviewers
@improbable-valentyn @m-samiec

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp [hunk 1]
```diff
@@ -69,6 +69,12 @@ Worker_EntityId USpatialPackageMapClient::AllocateEntityIdAndResolveActor(AActor
 	check(Actor);
 	checkf(bIsServer, TEXT("Tried to allocate an Entity ID on the client, this shouldn't happen."));
 
+	if (!IsEntityPoolReady())
+	{
+		UE_LOG(LogSpatialPackageMap, Error, TEXT("EntityPool must be ready when resolving an Actor: %s"), *Actor->GetName());
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp]: Is this an error? Couldn't this just happen in unfavourable initial level loading conditions?

Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp]: This happens on client. As we've discussed, the execution flow shouldn't even get to this stage. I added more checks to terminate the execution earlier.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp [hunk 2]
```diff
@@ -721,7 +721,10 @@ void USpatialReceiver::ReceiveActor(Worker_EntityId EntityId)
 			return;
 		}
 
-		PackageMap->ResolveEntityActor(EntityActor, EntityId);
+		if (!PackageMap->ResolveEntityActor(EntityActor, EntityId))
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: We should cleanup the actor and the channel here

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp (hunk 1 above)
Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialPackageMapClient.cpp]: There's a check above for running on server, perhaps that should be extended to return, rather then the inferred logic that the entity pool is setup.

File: SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp (hunk 2 above)
Comment (aleximprobable) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: Added
`			EntityActor->Destroy(true);
			Channel->Close(EChannelCloseReason::Destroyed);
`

Comment (m-samiec) [reply SpatialGDK/Source/SpatialGDK/Private/Interop/SpatialReceiver.cpp]: nit: Can we an additional log here for this explicit instance?

**Pull request merged**